h2o-py/h2o/frame.py

```diff
@@ -3182,9 +3182,16 @@ def cor(self, y=None, na_rm=False, use=None, method="Pearson"):
         assert_is_type(y, H2OFrame, None)
         assert_is_type(na_rm, bool)
         assert_is_type(use, None, "everything", "all.obs", "complete.obs")
-        if y is None:
-            y = self
-        if use is None: use = "complete.obs" if na_rm else "everything"
+        y_categorical = any(self.types[col_name] == "enum" for col_name in y)
```

Comment:
This seems incorrect - `y` H2OFrame not a list of columns. Also you can use `y.isfactor()` to check if it's categorical - the output is a list of boolean values in the same order as are the columns.